astropy/io/fits/file.py

```diff
@@ -487,22 +521,7 @@ def _open_filename(self, filename, mode, overwrite):
 
```

Comment:
The magic is read differently based on whether it's in `_open_fileobj` or `_open_filename` so I think it's best left as is.